tests/test_merge_all.py

```diff
@@ -274,3 +279,134 @@ def test_merge_all_on_non_overlapping_rasters_returns_first_raster():
     raster2 = make_test_raster(value=2, band_names=['blue'], affine=affine2, height=30, width=40)
     merged = merge_all([raster1, raster2], raster1.footprint())
     assert merged == raster1
+
+
+def test_merge_does_not_uncover_masked_pixels():
+    # See https://github.com/satellogic/telluric/issues/65
+    affine = Affine.translation(0, 2) * Affine.scale(1, -1)
+
+    rs_a = GeoRaster2(
+        image=np.ma.masked_array([
+            [
+                [100, 89],
+                [100, 89]
+            ],
+            [
+                [110, 99],
+                [110, 99]
+            ]
+        ], [
+            [
+                [False, True],
+                [False, True]
+            ],
+            [
+                [False, True],
+                [False, True]
+            ]
+        ], dtype=np.uint8),
+        affine=affine,
+        crs=WGS84_CRS,
+        band_names=['red', 'green'],
+    )
+
+    rs_b = GeoRaster2(
+        image=np.array([[
+            [0, 210],
+            [0, 210]
+        ]], dtype=np.uint8),
+        affine=affine,
+        crs=WGS84_CRS,
+        band_names=['green'],
+    )
+
+    expected_image = np.ma.masked_array([
+        [
+            [100, 89],
+            [100, 89]
+        ],
+        [
+            [110, 99],
+            [110, 99]
+        ]
+        ], [
+            [
+                [False, True],
+                [False, True]
+            ],
+            [
+                [False, True],
+                [False, True]
+            ]
+        ], dtype=np.uint8)
+
+    result = merge_all([rs_a, rs_b], rs_a.footprint()).limit_to_bands(['red', 'green'])
+
+    assert_array_equal(np.ma.filled(result.image, 0), np.ma.filled(expected_image, 0))
+    assert_array_equal(result.image.mask, expected_image.mask)
+
+
+def test_merge_all_non_overlapping_covers_all():
+    # See https://github.com/satellogic/telluric/issues/65
+    affine = Affine.translation(0, 2) * Affine.scale(1, -1)
+
+    rs1 = GeoRaster2(
+        image=np.array([[
+            [100, 0],
+            [100, 0]
+        ]], dtype=np.uint8),
+        affine=affine,
+        crs=WGS84_CRS,
+        band_names=['red'],
+    )
+
+    rs2 = GeoRaster2(
+        image=np.array([[
+            [110, 0],
+            [110, 0]
+        ]], dtype=np.uint8),
+        affine=affine,
+        crs=WGS84_CRS,
+        band_names=['green'],
+    )
+
+    rs3 = GeoRaster2(
+        image=np.array([[
+            [0, 200],
+            [0, 200]
+        ]], dtype=np.uint8),
+        affine=affine,
+        crs=WGS84_CRS,
+        band_names=['red'],
+    )
+
+    rs4 = GeoRaster2(
+        image=np.array([[
+            [0, 210],
+            [0, 210]
+        ]], dtype=np.uint8),
+        affine=affine,
+        crs=WGS84_CRS,
+        band_names=['green'],
+    )
+
+    expected_image = np.ma.masked_array([
```

Comment:
it looks like it assumes `0` is no data, or I'm wrong?

Reply:
Yes, it does assume it because it's the default behavior of `GeoRaster2.__init__`:

https://github.com/satellogic/telluric/blob/e315933a5dcb9072db11858fb047eeafe55297e3/telluric/georaster.py#L319